core/src/main/java/org/elasticsearch/common/xcontent/FromXContentBuilder.java

```diff
@@ -26,6 +26,7 @@
 /**
  * Indicates that the class supports XContent deserialization.
  */
+@FunctionalInterface
 public interface FromXContentBuilder<T> {
```

Comment:
Maybe use `NamedXContentRegistry.FromXContent` instead. Maybe extract it into a top level class? Remember that `ParseFieldMatcher` is being removed. If you want to keep the signatures the same you can use the static `ParseFieldMatcher.EMPTY` or remove all usages.